src/library/searchqueryparser.cpp

```diff
@@ -321,3 +353,96 @@ bool SearchQueryParser::queryIsLessSpecific(const QString& original, const QStri
     }
     return false;
 }
+
+QString SearchQueryParser::parseDateAddedArgument(const QString argRaw,
+        bool* special) const {
+    *special = false;
+    qDebug() << "   .";
+    qDebug() << "   .";
+    qDebug() << "   parseDateAddedArgument";
+    qDebug() << "   .";
+    QString date;
+    QString query;
+
+    // First, try to find literal date args
+    // TODO Add parser for 'lastNN[days|weeks|months|years]'
+    QRegularExpressionMatch wordMatch = kDateWordRegex.match(argRaw);
+    if (wordMatch.hasMatch()) {
+        QString range = wordMatch.captured(2);
+        qDebug() << "   .";
+        qDebug() << "   .";
+        qDebug() << "   wordmatch";
+        if (range == "week") {
+            date = "datetime('now', '-1 weeks')";
+        } else if (range == "month") {
+            date = "datetime('now', '-1 months')";
+        } else { // "year"
+            date = "datetime('now', '-1 years')";
+        }
+        *special = true;
+        query = QString(" > %1").arg(date);
+        return query;
+    } else {
+        qDebug() << "   .";
+        qDebug() << "   .";
+        qDebug() << "   no wordMatch";
+        qDebug() << "   .";
+    }
+
+    // Second, try to find operator + numerical args
+    QRegularExpressionMatch opMatch = kDateOperatorRegex.match(argRaw);
+    const QString op = opMatch.captured("op");
+    const QString arg = opMatch.captured("arg") + opMatch.captured(3);
+    if (!opMatch.hasMatch()) {
+        qDebug() << "   no opMatch";
+        return QString();
+    }
+    qDebug() << "   .";
+    qDebug() << "   hasOpMatch" << opMatch.hasMatch();
+    qDebug() << "     op: " << op;
+    qDebug() << "     arg:" << arg;
+    qDebug() << "   .";
+    qDebug() << "   .";
+
+    QRegularExpressionMatch yyyymmddMatch = kDateYyyyMmDdRegex.match(arg);
+    QRegularExpressionMatch yyyymmMatch = kDateYyyyMmRegex.match(arg);
+    QRegularExpressionMatch yyyyMatch = kDateYyyyRegex.match(arg);
+
+    // Search numerical dates, fill the day and/or month depending on the operator.
+    // Note we don't validate 'yyyy-mm-dd'
```

Comment:
I think real query examples as source code comments will help to understand the code.